### PR TITLE
Add Helm option to not include CRD

### PR DESCRIPTION
**Pull request by inteon**
TLDR; Adds the option to not install the CRD as part of the Helm chart. The CRD must still be installed separately.

If you install the approver-policy Helm chart with the `crds.enabled: true` option, uninstalling the Helm chart will result in also uninstalling the CRDs. This will trigger the CRs to be garbage collected and deleted too.
To prevent uninstalling the CRDs when the Helm chart is uninstalled, the CRDs can be installed separately. With the change proposed in this PR, this can be done as follows:

```bash
$ helm template cert-manager-approver-policy jetstack/cert-manager-approver-policy | \
        yq e '. | select(.kind == "CustomResourceDefinition")' \
       > approver-policy-crds.yaml

$ kubectl apply --server-side -f approver-policy-crds.yaml

$ helm upgrade cert-manager-approver-policy jetstack/cert-manager-approver-policy \
        -i -n cert-manager \
        --wait \
        --set crds.enabled=false
```

See https://github.com/cert-manager/trust-manager/pull/102#issuecomment-1400131204 for the equivalent change in trust-manager.

### CODE REVIEW

File: deploy/charts/approver-policy/values.yaml [hunk 1]
```diff
@@ -109,3 +109,6 @@ commonLabels: {}
 # -- Optional allow custom annotations to be placed on cert-manager-approver pod
 podAnnotations: {}
 
+crds:
+  # -- Whether or not to install the crds.
+  enabled: true
```
Comment (erikgb) [inline deploy/charts/approver-policy/values.yaml:114]: Any reason for using a value different from the one used in cert-manager, ref. https://cert-manager.io/docs/installation/helm/#helm-installation?

Comment (inteon) [reply deploy/charts/approver-policy/values.yaml:114]: It's based on trust-manager: https://github.com/cert-manager/trust-manager/blob/main/deploy/charts/trust-manager/values.yaml#L184-L186

Comment (erikgb) [reply deploy/charts/approver-policy/values.yaml:114]: Thanks, makes sense!

**Review by erikgb — COMMENTED** [review]
I think

https://github.com/cert-manager/approver-policy/blob/c459a137efbbe435f3033a3aa751b63da8cbc552/Makefile#L181

must be updated. But why are we changing the location of the CRD? Can we Helm template it in place?

**Comment by inteon** [comment]
I moved the crd because it was very confusing (instead of placing the crds in `templates/` or `crds/` it was placed in `templates/crds/`).

**Review by wallrj — CHANGES_REQUESTED** [review]
@inteon For the benefit of the changelog in the next release, please write *why* you're adding this option. Not being a pain, I honestly can't remember. What problem does it solve?

The documentation in the chart README does not say anything about why you might want to omit the CRDs. And the https://cert-manager.io/docs/policy/approval/approver-policy/ document doesn't talk about it either.

You mentioned that the option is based on a similar option in trust-manager, but I can't find any reason for the option there either:

```
$ helm inspect readme jetstack/trust-manager | grep crds
| crds.enabled | bool | `true` | Whether or not to install the crds. |
```

**Comment by inteon** [comment]
@wallrj same reason as https://github.com/cert-manager/trust-manager/pull/102

**Comment by wallrj** [comment]
> @wallrj same reason as [cert-manager/trust-manager#102 (comment)](https://github.com/cert-manager/trust-manager/pull/102#issuecomment-1400131204)

Ok. Then put something in the PR title and description so that whoever next releases approver-policy knows what to put in the release notes.

Here's my suggestion:
> Platform engineer needs to uninstall approver-policy without deleting all the CertificateRequestPolicy resources in the cluster. For example, when migrating from a vanilla approver-policy to an approver-policy "plugin", such as approver-policy-enterprise.
> In this PR we have partially solved this problem by allowing the approver-policy chart be installed without the CRDS,
> using the new `crds.enabled: false` value. This way the the platform engineer can uninstall the vanilla approver-policy chart and then install an approver-policy plugin chart without interfering with the CertificateRequestPolicy resources in the cluster.
> 
> How to extract the CRDs from the chart and install them manually, is currently currently left as an exercise for the platform engineer.
> In a follow up PR, in https://github.com/cert-manager/makefile-modules, we will implement a new `make release` feature to publish the CRDs to a separate downloadable manifest file attached to the GitHub release.
> We will also add documentation explaining why and how production ready deployments should install the CRDs before installing the chart.

**Comment by inteon** [comment]
> > @wallrj same reason as [cert-manager/trust-manager#102 (comment)](https://github.com/cert-manager/trust-manager/pull/102#issuecomment-1400131204)
> 
> Ok. Then put something in the PR title and description so that whoever next releases approver-policy knows what to put in the release notes.
> 
> Here's my suggestion:
> 
> > Platform engineer needs to uninstall approver-policy without deleting all the CertificateRequestPolicy resources in the cluster. For example, when migrating from a vanilla approver-policy to an approver-policy "plugin", such as approver-policy-enterprise.
> > In this PR we have partially solved this problem by allowing the approver-policy chart be installed without the CRDS,
> > using the new `crds.enabled: false` value. This way the the platform engineer can uninstall the vanilla approver-policy chart and then install an approver-policy plugin chart without interfering with the CertificateRequestPolicy resources in the cluster.
> > How to extract the CRDs from the chart and install them manually, is currently currently left as an exercise for the platform engineer.
> > In a follow up PR, in https://github.com/cert-manager/makefile-modules, we will implement a new `make release` feature to publish the CRDs to a separate downloadable manifest file attached to the GitHub release.
> > We will also add documentation explaining why and how production ready deployments should install the CRDs before installing the chart.

I updated the PR details with the exact instructions necessary to install the CRDs separately and why you would like to do this.

**Comment by wallrj** [comment]
I think it was you who told me about `"helm.sh/resource-policy": keep` in [Tell Helm Not To Uninstall a Resource](https://helm.sh/docs/howto/charts_tips_and_tricks/#tell-helm-not-to-uninstall-a-resource); why not do that instead?

If the instructions are to use `yq` to get only the CRDs, then why not solve this problem by also using `yq` to get everything **but** the CRDs?
```sh
helm template cert-manager-approver-policy jetstack/cert-manager-approver-policy \
    | yq e '. | select(.kind != "CustomResourceDefinition")'
```

If the goal is to allow quick and simple "default" installation of both CRDs and other stuff, for users who are evaluating approver-policy, why don't we put the CRDs in the `crds/` directory as recommended by Helm.
And then write upgrade instructions for production users telling them that they will have to "upgrade" the CRDs separately?

**Comment by inteon** [comment]
fixed in https://github.com/cert-manager/approver-policy/pull/376
